src/mds/Server.cc

```diff
@@ -1046,6 +1102,93 @@ void Server::respond_to_request(MDRequestRef& mdr, int r)
     mdr->internal_op_finish->complete(r);
     mdcache->request_finish(mdr);
   }
+
+  // add here to avoid counting ops multiple times (e.g., locks, loading)
+  switch(mdr->client_request->get_op()) {
```

Comment:
Oops, missed this in review -- mdr->client_request may be null (see conditional above), so this breaks when internal ops are executed.